components/browser/state/src/test/java/mozilla/components/browser/state/action/EngineActionTest.kt

```diff
@@ -81,4 +84,36 @@ class EngineActionTest {
         assertNotNull(engineState().engineObserver)
         assertEquals(engineObserver, engineState().engineObserver)
     }
+
+    @Test
+    fun `PurgeHistoryAction - Removes state from sessions without history`() {
+        val tab1 = createTab("https://www.mozilla.org").copy(
+            engineState = EngineState(engineSession = null, engineSessionState = mock())
+        )
+
+        val tab2 = createTab("https://www.firefox.com").copy(
+            engineState = EngineState(engineSession = mock(), engineSessionState = mock())
+        )
+
+        val customTab1 = createCustomTab("http://www.theverge.com").copy(
+            engineState = EngineState(engineSession = null, engineSessionState = mock())
+        )
+
+        val customTab2 = createCustomTab("https://www.google.com").copy(
+            engineState = EngineState(engineSession = mock(), engineSessionState = mock())
+        )
+
+        val store = BrowserStore(BrowserState(
+            tabs = listOf(tab1, tab2),
+            customTabs = listOf(customTab1, customTab2)
+        ))
+
+        store.dispatch(EngineAction.PurgeHistoryAction).joinBlocking()
```

Comment:
Do we also need to add `dispatcher.advanceUntilIdle()` or  `store.waitUntilIdle()` to avoid flakiness on the following checks?

Reply:
In this test with use `joinBlocking()` on the dispatch call. After that completed the action was completely processed. Since there's no async middleware involved here, this is enough.

Reply:
👍🏽